msl-states.h

```diff
@@ -0,0 +1,171 @@
+#ifndef MSL_STATES_H
+#define MSL_STATES_H
+
+#include <stdint.h>
+#include "msl-commands.h"
+#include "msl-states.h"
+
+// Initialitiation
+
+// to state E2+E3 if !AINI
+//          C8    if AINI
+
+static uint8_t AINI = 0; // stub value to test condition logic
+
+uint8_t state_80_TI06_CU01(struct ge *ge) { return !AINI; }
+uint8_t state_80_TI06_CU03(struct ge *ge) { return AINI; }
+uint8_t state_80_TI06_CU05(struct ge *ge) { return !AINI; }
+
+struct msl_timing_chart state_80[] = {
+    { TO30, CI19, 0 },
+    { TO30, CO96, 0 },
+    { TO30, CO97, 0 },
+    { TO40, CO00, 0 },
+    { TO40, CO02, 0 },
+    { TO50, CI32, 0 },
+    { TI05, CI05, 0 },
+    { TI05, CI08, 0 },
+    { TI06, CI76, 0 },
+    { TI06, CI80, 0 },
+    { TI06, CI81, 0 },
+    { TI06, CI82, 0 },
+    { TI06, CU01, state_80_TI06_CU01 },
+    { TI06, CU03, state_80_TI06_CU03 },
+    { TI06, CU05, state_80_TI06_CU05 },
+    { TI06, CU06, 0 },
+};
+
+// Alpha phase
+
+// (to state F0 if RINT & !FA06
+//           E0 if !RINT | FA06)
+
+uint8_t state_E2_E3_TO80_CI89(struct ge *ge) { return 0; }
+uint8_t state_E2_E3_TI06_CI82(struct ge *ge) { return 0; }
+uint8_t state_E2_E3_TI06_CU04(struct ge *ge) { return 0; }
+
+struct msl_timing_chart state_E2_E3[] = {
+    { TO10, CO10, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO40, CO02, 0 },
+    { TO70, CI67, 0 },
+    { TO70, CI62, 0 },
+    { TO80, CI89, state_E2_E3_TO80_CI89 },
+    { TI05, CI08, 0 },
+    { TI06, CI80, 0 },
+    { TI06, CI82, state_E2_E3_TI06_CI82 },
+    { TI06, CI83, 0 },
+    { TI06, CU04, state_E2_E3_TI06_CU04 },
+    { TI06, CU10, 0 },
+    { TI06, CU11, 0 },
+};
+
+// to state E4    if FO06 | FO07
+//          64+65 if !(FO06 | FO07)
+
+uint8_t state_E0_TO70_CI60(struct ge *ge) { return 0; }
+uint8_t state_E0_TI06_CU17(struct ge *ge) { return 1; }
+
+struct msl_timing_chart state_E0[] = {
+    { TO10, CO12, 0 },
+    { TO10, CO41, 0 },
+    { TO25, CO30, 0 },
+    { TO40, CO00, 0 },
+    { TO50, NONE, 0 }, // NO -> BO
```

Comment:
This is an intrinsic master pulse event, and will be executed on `on_TO50` through `do_pulse`. Please remove